### PR TITLE
add tensorflow-macos support

### PR DESCRIPTION
**Pull request by slayerjain**
ref - https://github.com/huggingface/datasets/issues/2068

### CODE REVIEW

File: src/datasets/config.py [hunk 1]
```diff
@@ -65,6 +65,7 @@
             "tensorflow",
             "tensorflow-cpu",
             "tensorflow-gpu",
+            "tensorflow-macos",
```
Comment (albertvillanova) [inline src/datasets/config.py:68]:
```suggestion
```

File: src/datasets/config.py [hunk 2]
```diff
@@ -65,6 +65,7 @@
...
             "tf-nightly",
             "tf-nightly-cpu",
             "tf-nightly-gpu",
```
Comment (albertvillanova) [inline src/datasets/config.py:71]:
```suggestion
            "tf-nightly-gpu",
            "intel-tensorflow",
            "tensorflow-rocm",
            "tensorflow-macos",
```

**Review by albertvillanova — CHANGES_REQUESTED** [review]
Hi @slayerjain, thanks for updating this.

Would you mind to add the other TensorFlow builds as well, so that we are completely aligned with `transformers`?

**Comment by slayerjain** [comment]
@albertvillanova done!

**Review by albertvillanova — APPROVED** [review]
Thank you for your Pull Request.

**Pull request merged**